lib/manageiq/providers/ibm_cloud/cloud_tools/common/proxy.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ManageIQ
+  module Providers
+    module IbmCloud
+      module CloudTools
+        module Common
+          # Set proxy configurations dynamically.
+          class Proxy
+            def initialize
+              @keys = %i[address port username password headers]
+              @keys.each { |key| self.class.attr_accessor(key) }
+            end
+
+            # Add a server configuration.
+            # @return [Proxy]
+            def server(address:, port:)
+              @address = address
+              @port = port
+              self
+            end
+
+            # Add a proxy authentication to the server configuration.
+            # @return [Proxy]
+            def authentication(username:, password:)
+              @username = username
+              @password = password
+              self
+            end
+
+            # Convert non-nil keys into a hash.
+            # @return [Hash]
+            def to_hash
+              data = {}
+              @keys.each do |key|
+                variable = instance_variable_get("@#{key}")
+                next if variable.nil?
+
+                data[key] = variable # unless variable.nil?
+              end
+              data
```

Comment:
Slightly simpler

```suggestion
              @keys.each_with_object({}) { |k, h| h[k] = instance_variable_get("@#{key}") }.compact
```

Reply:
Reimplemented in conjunction with suggestions in https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/137#discussion_r577167542